keps/sig-api-machinery/1929-built-in-default/README.md

```diff
@@ -192,29 +194,32 @@ values in the merge patch are given special meaning to indicate the
 removal of existing values in the
 target."(link)[https://tools.ietf.org/rfc/rfc7386.txt#:~:text=Null%20values%20in%20the%20merge]
 
-On the other hand, while `{field: null}` and `{}` probably have the same
+On the other hand, while `{"field": null}` and `{}` probably have the same
 meaning, they do not "look" the same and are hard to compare for
 clients. They are also currently treated differently by Kubernetes for
 defaulting purposes.
 
-We propose the creation of a new CRD flag: `preserveNullValues` which
-will default to `true`. This flag will be used as a ratcheting mechanism
-as we migrate to this new semantic throughout all Kubernetes types
-(which would be how all types eventually work). When false, the
-apiserver will use new semantics and remove all null values from CR
-objects. When true, the existing and deprecated semantics will be kept.
-That new flag is following the same logic as "preserveUnknownFields" and
-the new operation will happen using the same mechanisms right after
-pruning.
+Currently, null values trigger an error when nullable is false and don't
+get defaulted if a default is available. We propose to start defaulting
+fields with null values if a default is available, or the field will be
+removed if no default exists, but only when the openapi nullable flag is
+missing or false. This means that will start accepting inputs that were
+rejected in the past. There is code that would work with a new server
+but not with an old server. We assume that existing APIs that use
+nullable are happy with null values. Also, people can remove the
+nullable attribute if they want null values to start being removed. This
+will help aligning CRDs and built-in types (there are no ways to set
+nullable to true for built-in types).
```

Comment:
Maybe worth a short node:

We will remove null fields also from `map[string]T`s, i.e. for type that have `additionalProperties` in their schema instead of `properties`.